### PR TITLE
Replace find with grep -rl in update scripts

### LINKED ISSUE #12735: hack/verify-generated-defaulters.sh broken on base_ami
Seen in many of the latest runs like [here](https://ci.openshift.redhat.com/jenkins/view/AMIs/job/base_ami/2659/consoleFull):

```
===== Verifying Generated Defaulters =====
FAILURE: Verification of defaulters failed:
[WARNING] No compiled `gendefaults` binary was found. Attempting to build one using:
[WARNING]   $ hack/build-go.sh tools/gendefaults
++ Building go targets for linux/amd64: tools/gendefaults
++ Placing binaries
/data/src/github.com/openshift/origin/hack/build-go.sh took 3 seconds
[ERROR] PID 23313: hack/update-generated-defaulters.sh:17: `INPUT_DIRS=$(
[ERROR]   find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | 	xargs grep --color=never -l '+k8s:defaulter-gen=' | 	xargs -n1 dirname | 	sed "s,^\.,${PREFIX}," | 	sort -u | 	paste -sd,
[ERROR] )` exited with status 123.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/update-generated-defaulters.sh:17: `INPUT_DIRS=$(
[INFO]   find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | 	xargs grep --color=never -l '+k8s:defaulter-gen=' | 	xargs -n1 dirname | 	sed "s,^\.,${PREFIX}," | 	sort -u | 	paste -sd,
[INFO] )`
[INFO]   Exiting with code 123.
```

@ncdc pass this on to someone to own please

### PR DESCRIPTION
**Pull request by ncdc**
Fixes #12735 

@stevekuznetsov @smarterclayton

### CODE REVIEW

**Comment by ncdc** [comment]
[test]

File: hack/update-generated-defaulters.sh [hunk 1]
```diff
@@ -5,18 +5,19 @@ os::build::setup_env
...
+PREFIX=github.com/openshift/origin/
 
 INPUT_DIRS=$(
-  find . -not  \( \( -wholename '*/vendor/*' \) -prune \) -name '*.go' | \
-	xargs grep --color=never -l '+k8s:defaulter-gen=' | \
-	xargs -n1 dirname | \
-	sed "s,^\.,${PREFIX}," | \
-	sort -u | \
-	paste -sd,
+  cd ${OS_ROOT}
```
Comment (stevekuznetsov) [inline hack/update-generated-defaulters.sh]: The contract from `hack/lib/init.sh` is that you are in `$OS_ROOT` -- no need for `cd`

File: hack/update-generated-defaulters.sh [hunk 2]
```diff
@@ -25,8 +26,8 @@ EXTRA_PEER_DIRS=$(
 )
 
 gendefaults \
-	--logtostderr \
-	--output-base="${GOPATH}/src" \
-	--input-dirs ${INPUT_DIRS[@]} \
+  --logtostderr \
+  --output-base="${GOPATH}/src" \
+  --input-dirs ${INPUT_DIRS[@]} \
```
Comment (stevekuznetsov) [inline hack/update-generated-defaulters.sh]: These vars aren't arrays so I don't really understand the `[@]`. If you do make them arrays you need to quote this so it breaks on array elements and not spaces, if you don't make them arrays you need to quote this so it's always guaranteed to be one arg to `--input-dirs`

File: hack/update-generated-openapi.sh [hunk 3]
```diff
@@ -5,33 +5,33 @@ os::build::setup_env
 
 os::util::ensure::built_binary_exists 'genopenapi'
 
-ORIGIN_PREFIX=github.com/openshift/origin
+ORIGIN_PREFIX=github.com/openshift/origin/
```
Comment (stevekuznetsov) [inline hack/update-generated-openapi.sh]: this is `"${OS_GO_PACKAGE}/"`

**Comment by ncdc** [comment]
@stevekuznetsov updated, ptal

**Comment by ncdc** [comment]
Unnecessary, yes. But tabs are evil.

**Comment by stevekuznetsov** [comment]
> Unnecessary, yes. But tabs are evil.

Don't let @deads2k or @liggitt hear you say that.

**Comment by deads2k** [comment]
> Unnecessary, yes. But tabs are evil.

People who use spaces hate freedom. :)

**Comment by ncdc** [comment]
[merge]

**Pull request merged**